nirc_ehr/src/org/labkey/nirc_ehr/table/NIRC_EHRCustomizer.java

```diff
@@ -295,6 +312,72 @@ public boolean isEditable()
         ti.addColumn(ci);
     }
 
+    private void appendPairingUpdateCol(AbstractTableInfo ti)
+    {
+        String name = "updatePairing";
+        String label = "Update Pairing";
+        if (ti.getColumn(name) != null)
+            return;
+
+        var ci = new WrappedColumn(ti.getColumn("Id"), name);
+        ci.setDisplayColumnFactory(new DisplayColumnFactory()
+        {
+            @Override
+            public DisplayColumn createRenderer(final ColumnInfo colInfo)
+            {
+                return new DataColumn(colInfo){
+
+                    @Override
+                    public void renderGridCellContents(RenderContext ctx, Writer out) throws IOException
+                    {
+                        ActionURL linkAction = new ActionURL("ehr", "dataEntryForm", ti.getUserSchema().getContainer());
+                        if (!ti.getUserSchema().getContainer().hasPermission(ti.getUserSchema().getUser(), EHRBehaviorEntryPermission.class))
+                            return;
+
+                        linkAction.addParameter("formType", "Pairing Observations");
+
+                        String taskId = (String)ctx.get("taskid");
+                        if (StringUtils.isNotEmpty(taskId))
+                        {
+                            linkAction.addParameter("taskid", taskId);
+                            String href = linkAction.toString();
+                            out.write(PageFlowUtil.link(label).href(href).target("_blank").toString());
+                        }
+                    }
+
+                    @Override
+                    public void addQueryFieldKeys(Set<FieldKey> keys)
+                    {
+                        super.addQueryFieldKeys(keys);
```

Comment:
You need to add taskid here, otherwise this will only be available if taskid happens to be in the current view.

Reply:
Added.